floater/generators.py

```diff
@@ -50,6 +71,39 @@ def get_rectmesh(self):
 
         return np.meshgrid(self.x, self.y)
 
+
+    def get_hexcoords(self, mask, mask_grid_x, mask_grid_y):
+        """Get the coordinates of float positions taking into account the land mask.
+
+        RETURNS
+        -------
+        floats_ocean: np.ndarray
+            1D array of float coordinate subarrays: [lat, lon] 
+        """
+
+        xx, yy = self.get_rectmesh()
+        # modify to be even-R horizontal offset
+        xx[::2] += self.dx/4
+        xx[1::2] -= self.dx/4
+
+        mask_lon = mask_grid_x
+        mask_lat = mask_grid_y 
+        mask_geo = np.dstack(np.meshgrid(mask_lon, mask_lat)).reshape(-1, 2) # fast cartesian product
+        mask_bool_flat = mask.ravel()
+        mask_xyz = geo_to_xyz(mask_geo) 
+        mask_tree = KDTree(mask_xyz) # a KDTree of the mask data in xyz form 
+        floats_geo = np.transpose([xx.ravel(), yy.ravel()]) # uniform hexagonal tiling   
+        queries_xyz = geo_to_xyz(floats_geo)
+
+        # search for 4 nearest neighbors
+        neighbor_indices = mask_tree.query(queries_xyz, k=4, return_distance=False) 
+        neighbor_bools = np.take(mask_bool_flat, neighbor_indices.ravel(), axis=0) # True -> neighbor is ocean
+        neighbor_bools = np.reshape(neighbor_bools, (-1, 4)) # reconstruct subarrays of 4 neighbors 
+        ocean_bools = np.any(neighbor_bools, axis=1) # OR statement on each subarray of 4 neighbors
+        floats_ocean = floats_geo[np.nonzero(ocean_bools.astype('int'))]
+
+        return floats_ocean 
+
```

Comment:
My thinking was that this code does not return a `np.meshgrid`, so it might be confusing to the user if we incorporated it into `get_hexmesh`. Is that a concern? If not, I'm happy to make this change.